demos/sev-demo/README.md

```diff
@@ -0,0 +1,136 @@
+
+## Rootfs
+ 
+Create the rootfs as you normally would for CCv0, but be sure to specify set `AA_KBC` to `offline_sev_kbc` so that the AA is built with the correct key broker client. 
+
+```
+sudo make USE_DOCKER=yes DISTRO=ubuntu AGENT_INIT=yes DOCKER_RUNTIME=runc AA_KBC=offline_sev_kbc SKOPEO_UMOCI=yes rootfs
+```
+
+You will need to add the agent configuration file to the initrd. Again, this is a typical CCv0 requirement, but you will need to specify parameters for you KBC. Specifically be sure to add this parameter.
+```
+aa_kbc_params = "offline_sev_kbc::null"
+```
+
+The `offline_sev_kbc` uses the `efi_secret` kernel module to expose injected secrets to the filesystem. You'll need to add the kernel module to the initrd. You can do this by setting the `KERNEL_MODULES_DIR` when you build your rootfs or you can copy the kernel module directly into the rootfs that is created. Either way, your rootfs should look something like this. 
+```
+$ tree ubuntu_rootfs/lib/modules
+lib/modules
+└── 5.15.0-rc5+
+    ├── kernel
+    │   └── drivers
+    │       └── virt
+    │           └── coco
+    │               └── efi_secret
+    │                   └── efi_secret.ko
+    ├── modules.alias
+    ├── modules.alias.bin
+    ├── modules.builtin
+    ├── modules.builtin.alias.bin
+    ├── modules.builtin.bin
+    ├── modules.builtin.modinfo
+    ├── modules.dep
+    ├── modules.dep.bin
+    ├── modules.devname
+    ├── modules.order
+    ├── modules.softdep
+    ├── modules.symbols
+    └── modules.symbols.bin
+```
+
+Make sure that the module matches your kernel version. Speaking of which, `efi_secret` also requires some minor kernel patches. These patches and the module are not currently upstream. You will need to find Dov Murik's EFI secret patches on the kernel mailing list and build a kernel with them. This should also give you the correct version of the `efi_secret` module.  
+
+
+## Kata Runtime
+
+Since SEV and SEV-ES use pre-attestation driven from outside of the guest, some changes are required for the Kata runtime. These changes are not currently upstream. You will need to build your Kata runtime with [this PR](https://github.com/kata-containers/kata-containers/pull/3025). This will work best with QEMU 6.2.0rc0. Newer versions of QEMU may require you to set `kernel-hashes=on` via the Kata runtime.
+
+You will also need to set some parameters in the Kata configuration file usually located at `/etc/kata-containers/configuration.toml`.
+
+First, you'll need to enable SEV via the `confidential_guest` parameter.
+```
+# Enable confidential guest support.
+# Toggling that setting may trigger different hardware features, ranging
+# from memory encryption to both memory and CPU-state encryption and integrity.
+# The Kata Containers runtime dynamically detects the available feature set and
+# aims at enabling the largest possible one.
+# Default false
+confidential_guest = true
```

Comment:
You might want to consider either using `sed` (or the `crudini` tool) for setting these options. Admittedly you won't then show the comments, but those might change so hard-coding them here may confuse users more than it helps when they diverge.

Reply:
I'm not a huge fan of guides that use `sed`. I tend to just blindly run the command and I suspect other people do too.

Reply:
I'm most in favor of enabling k8s annotation for these configuration options so that they can be specified in the pod yaml. Then we wouldn't need to set global defaults (although, we'd probably still want instructions to do so)

Reply:
I guess it's a personal thing. But we should come up with an agreed policy for points like this. Kata uses the following lightweight guidance doc:

https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md.